pkg/service/worker_service_integration_test.go

```diff
@@ -0,0 +1,175 @@
+package service_test
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+
+	"github.com/MinhNghiaD/jobworker/api/client"
+	"github.com/MinhNghiaD/jobworker/api/worker/proto"
+	"github.com/MinhNghiaD/jobworker/pkg/service"
+	"github.com/sirupsen/logrus"
+)
+
+func TestSimulation(t *testing.T) {
```

Comment:
I don't think this level of testing is necessary for this challenge, you can stick to the basics.

As an aside, I don't think this is a great test. It seems like it will never actually fail, and if it could I personally prefer to avoid this level of randomness in tests because they tend to be flaky.

Reply:
This test is aim for race detection and go routine leak, the previous unit tests already cover the the behaviours of functionalities

Reply:
Good point, some concurrency test like this is good for race detection. But I think if there is an unexpected error in one of your calls, the test should still fail instead of writing a log that will likely be ignored

Reply:
Yes but I think that should be covered in more detail test cases with some specific use cases and predicted results. It hards to control errors in a random test like this, as you suggested very flaky. The main purpose of this test is only to see if there is any data race, routine leak for is the service will crashed under concurrent requests. I think we should keep it that way and add more test if needed rather than put all in on test

Reply:
You may want to add a comment with this rationale to the test. Also See my other comment re: randomness/repeatability.